### ISSUE TITLE
Suite reruns few times

### ISSUE THREAD

**Issue by Glueon** [joeyespo/pytest-watch#12]
If it's important I am using pytest-watch with https://github.com/pytest-dev/pytest-incremental.

When I update any source file sometimes pytest-watch reruns test twice, sometimes even three times. Is it a problem or a feature?

**Comment by joeyespo** [comment]
Interesting. Does it happen when not using the incremental plugin?

If incremental works by manipulating the files system, that could be why.

**Comment by Glueon** [comment]
Actually I realized it happens even I use `loopfail` feature from xdist. The only difference is that with xdist I've neve seen test reruning 3 time in row :) If you use the same code base as xdist then maybe it's a xdists's bug.

**Comment by blueyed** [comment]
What editor are you using?
It might create a temporary file first, and then move it etc.

**Comment by Glueon** [comment]
Just VIM.  `backupdir`, `undodir` and a `directory` settings are all located in the subderictories under ~/.vim/. So they shouldn't interfear.

**Comment by aldanor** [comment]
@joeyespo Seems like the same problem I've been seeing :)

**Comment by blueyed** [comment]
@Glueon 
You could add a `print(event)` to `ChangeHandler.on_any_event` to see what happens.  It appears to work fine for me - there's only one event for me:

    <FileModifiedEvent: src_path='/tmp/foo/file'>

**Comment by aldanor** [comment]
@blueyed @Glueon I already fixed it a few hours ago in #11 PR

**Comment by Glueon** [comment]
@aldanor 
Well seems that it's still an issue. As I said I am using `docker-compose run --rm my_container pwt` to run the tests. My code folder is mounted using volume from a host machine. So when I save a file in VIM it runs tests exactly 3 times.

**Comment by aldanor** [comment]
@Glueon 
Have you tried the code from #11 branch (it hasn't been merged into master yet)?

**Comment by Glueon** [comment]
@aldanor 
Oh, snap. I thougt it's in the master branch. 
Your version fires uo 3 events: created, deleted and modified, but runs only once. These 3 event seem kind of strange, but as soon as it runs once I am totally ok with that.

**Comment by aldanor** [comment]
Yep, that's caused by atomic saves, hence we have to spool events manually

**Comment by Glueon** [comment]
@joeyespo As soon as these changes get merged yo can close this issue.

**Comment by blueyed** [comment]
Great!

@Glueon 
It's related to Vim's `backupcopy` setting.  You could use `set backupcopy=yes`.

**Comment by Glueon** [comment]
@blueyed

This will reduce 3 events to 1?

**Comment by blueyed** [comment]
Just for reference: here is a list how different editors handle this: https://github.com/guard/guard/wiki/Analysis-of-inotify-events-for-different-editors

**Comment by joeyespo** [comment]
@blueyed That's a fantastic reference. Thanks!

**Comment by joeyespo** [comment]
@Glueon Merged! Thanks for opening the issue.

**Comment by joeyespo** [comment]
FYI Released pytest-watch 3.0.0 today, which includes the merged changes from #11. Be sure to `pip install --upgrade pytest-watch` and let me know if something broke!
